### PR TITLE
OHRM5X-2052: Develop Assign Claim screen

### PR DESCRIPTION
**Pull request by DulsaraNethmin**

### CODE REVIEW

File: src/plugins/orangehrmClaimPlugin/Controller/AssignClaimRequestController.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        $id = $request->attributes->getInt('id');
+        $claimRequest = $this->getClaimService()->getClaimDao()->getClaimRequestById($id);
+        if (
+            !$claimRequest instanceof ClaimRequest ||
+            $this->getUserRoleManagerHelper()->isSelfByEmpNumber(
+                $claimRequest->getEmployee()->getEmpNumber()
+            ) ||
+            !$this->getUserRoleManagerHelper()->isEmployeeAccessible(
+                $claimRequest->getEmployee()->getEmpNumber()
+            )
```
Comment (ManulMax) [inline src/plugins/orangehrmClaimPlugin/Controller/AssignClaimRequestController.php:86]:
```suggestion
!$claimRequest instanceof ClaimRequest
            || $this->getUserRoleManagerHelper()->isSelfByEmpNumber($claimRequest->getEmployee()->getEmpNumber())
            || !$this->getUserRoleManagerHelper()->isEmployeeAccessible($claimRequest->getEmployee()->getEmpNumber())
```
Format code

File: src/plugins/orangehrmClaimPlugin/config/routes.yaml [hunk 2]
```diff
@@ -162,11 +162,22 @@ claim_view_claim:
   path: /claim/viewClaim
   controller: OrangeHRM\Claim\Controller\ViewMyClaimController::handle
   methods: [ GET ]
+
+claim_assign_claim:
+  path: /claim/assignClaim
+  controller: OrangeHRM\Claim\Controller\AssignClaimController::handle
+  methods: [GET]
```
Comment (ManulMax) [inline src/plugins/orangehrmClaimPlugin/config/routes.yaml:169]:
```suggestion
  methods: [ GET ]
```

File: src/plugins/orangehrmClaimPlugin/config/routes.yaml [hunk 3]
```diff
@@ -162,11 +162,22 @@ claim_view_claim:
...
+
 ##########################
 #        Files
 ##########################
 
 claim_attachment:
   path: /claim/requests/{requestId}/attachId/{attachId}
   controller: OrangeHRM\Claim\Controller\ClaimAttachmentController::handle
-  methods: [ GET ]
+  methods: [GET]
```
Comment (ManulMax) [inline src/plugins/orangehrmClaimPlugin/config/routes.yaml:183]:
```suggestion
 methods: [ GET ]
```

**Review by ManulMax — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmClaimPlugin/components/ClaimEmployeeAutocomplete.vue [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        {{ $t('general.past_employee') }}
+      </div>
+    </template>
+  </oxd-input-field>
+</template>
+
+<script>
+import {APIService} from '@ohrm/core/util/services/api.service';
+export default {
+  name: 'ClaimEmployeeAutocomplete',
```
[the remark this answers is not in the document]
Comment (DulsaraNethmin) [reply src/client/src/orangehrmClaimPlugin/components/ClaimEmployeeAutocomplete.vue:45]: need to stop suggesting admin's name.

File: src/client/src/orangehrmClaimPlugin/components/ClaimActionButtons.vue [hunk 5]
```diff
@@ -124,11 +129,13 @@ export default {
...
         .then(() => {
-          navigate(`/claim/submitClaim/id/${this.requestId}`);
+          this.isAssigned
+            ? navigate(`/claim/assignClaim/id/${this.requestId}`)
+            : navigate(`/claim/submitClaim/id/${this.requestId}`);
         });
     },
     onBack() {
-      navigate('/claim/submitClaim');
+      navigate('/claim/assignClaim');
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/components/ClaimActionButtons.vue:138]: Double check back button behaviour

File: src/client/src/orangehrmClaimPlugin/pages/assignClaim/AssignClaim.vue [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+      name: props.empName,
+    };
+  },
+
+  data() {
+    const statusMap = {
+      SUBMITTED: this.$t('time.submitted'),
+      APPROVED: this.$t('time.approved'),
+      REJECTED: this.$t('leave.rejected'),
+      CANCELLED: this.$t('leave.cancelled'),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmClaimPlugin/pages/assignClaim/AssignClaim.vue:169]: TODO:: we may have to move these status to `general` language group

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Comment by RajithaKumara** [comment]
Closing as this code goes in #1690 as well. Nethmin confirmed all changes are there in that PR.
cc @ManulMax

**Pull request closed without merging**